src/l500/l500-color.cpp

```diff
@@ -493,10 +493,44 @@ namespace librealsense
         return tags;
     }
 
-    std::vector<uint8_t> l500_color::get_raw_intrinsics_table() const
+    ivcam2::intrinsic_rgb l500_color::get_intrinsics_table() const
     {
-        AC_LOG( DEBUG, "RGB_INTRINSIC_GET" );
-        return _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+        // Get RAW data from firmware
+        std::vector< uint8_t > response_vec = _hw_monitor->send(command{ RGB_INTRINSIC_GET });
+
+        if (response_vec.size() == 0)
+            throw invalid_value_exception("Calibration data invalid,buffer size is zero");
+
+        auto resolutions_rgb_table_ptr = reinterpret_cast<const ivcam2::intrinsic_rgb *>(response_vec.data());
+
+        // Get full maximum size of the resolution array and deduct the unused resolutions size from it
+        size_t expected_size = sizeof( ivcam2::intrinsic_rgb )
+                                - ( ( MAX_NUM_OF_RGB_RESOLUTIONS
+                                - resolutions_rgb_table_ptr->resolution.num_of_resolutions )
+                                * sizeof( pinhole_camera_model ) );
+
+        // Validate table size
+        if (expected_size > response_vec.size())
```

Comment:
Also check for expected < actual, or just use !=
I think that since the table size is const it can all be replaced with check_calib<> above

Reply:
See my comment above: we can only use the bytes covered by the number of resolutions, but we're not guaranteed that there will be no extra information.

Reply:
The table has the reserved space for up to 5 tables, but only some of them will be active.
Yet the raw table size is predefined by FW spec, thus non-modifiable, is it?

Reply:
Like mentioned above, FW send extra bytes for debugging purpose. and it can change on each FW version.
+ we get different size of output for USB2/USB3